src/main/java/org/takes/rq/RqMultipart.java

```diff
@@ -236,15 +223,47 @@ public Base(final Request req) throws IOException {
                     break;
                 }
                 this.buffer.position(this.buffer.position() + 1);
-                requests.add(this.make(boundary));
+                requests.add(this.make(boundary, body));
             }
             return RqMultipart.Base.asMap(requests);
         }
         /**
+         * Init a map of params and values.
+         * @param req Request
+         * @return ConcurrentMap, emtpy if something wrong
+         * @throws IOException if fails
+         */
+        private ConcurrentMap<String, List<Request>> initMap(final Request req)
+            throws IOException {
+            final InputStream stream = new RqLengthAware(req).body();
+            this.initBuffer(stream);
+            final ReadableByteChannel body = Channels.newChannel(stream);
+            ConcurrentMap<String, List<Request>> result = new
+                ConcurrentHashMap<String, List<Request>>();
+            try {
+                result = this.buildRequests(req, body);
+            } finally {
+                body.close();
+                stream.close();
```

Comment:
@2686747 this is not a good idea, actually. what if we fail to close `body` and throw an exception? `stream` won't be closed ever?